ptzclient/client.go

```diff
@@ -0,0 +1,517 @@
+	var panSpeed, tiltSpeed, zoomSpeed float64
+	if sendSpeed {
+		panSpeed = getOptionalFloat64(cmd, "pan_speed", defaultPanSpeed)
+		tiltSpeed = getOptionalFloat64(cmd, "tilt_speed", defaultTiltSpeed)
+		zoomSpeed = getOptionalFloat64(cmd, "zoom_speed", defaultZoomSpeed)
+	}
+
+	// Validate position ranges (-1.0 to 1.0 for pan/tilt, 0.0 to 1.0 for zoom)
+	if panPos < -1.0 || panPos > 1.0 || tiltPos < -1.0 || tiltPos > 1.0 || zoomPos < 0.0 || zoomPos > 1.0 {
+		return nil, errors.New("position values must be within normalized range (-1 to 1 for pan/tilt, 0 to 1 for zoom)")
+	}
+
+	req := ptz.AbsoluteMove{
+		ProfileToken: profileToken,
+		Position: onvifxsd.PTZVector{
+			PanTilt: onvifxsd.Vector2D{
+				X:     panPos,
+				Y:     tiltPos,
+				Space: AbsolutePanTiltPositionGenericSpace,
+			},
+			Zoom: onvifxsd.Vector1D{
+				X:     zoomPos,
+				Space: AbsoluteZoomPositionGenericSpace,
+			},
+		},
+	}
+
+	if sendSpeed {
+		// Validate speeds if they are being sent
+		if panSpeed < 0.0 || panSpeed > 1.0 || tiltSpeed < 0.0 || tiltSpeed > 1.0 || zoomSpeed < 0.0 || zoomSpeed > 1.0 {
+			return nil, errors.New("speed values (pan_speed, tilt_speed, zoom_speed) must be between 0.0 and 1.0 for absolute move")
+		}
+		req.Speed = onvifxsd.PTZSpeed{
+			PanTilt: onvifxsd.Vector2D{
+				X:     panSpeed,
+				Y:     tiltSpeed,
+				Space: AbsolutePanTiltPositionGenericSpace,
+			},
+			Zoom: onvifxsd.Vector1D{
+				X:     zoomSpeed,
+				Space: AbsoluteZoomPositionGenericSpace,
+			},
+		}
+		s.logger.Debugf("Sending AbsoluteMove (P: %.3f, T: %.3f, Z: %.3f) with Speed (X: %.2f, Y: %.2f, Z: %.2f) for profile %s...",
+			panPos, tiltPos, zoomPos, panSpeed, tiltSpeed, zoomSpeed, profileToken)
+	} else {
+		s.logger.Debugf("Sending AbsoluteMove (P: %.3f, T: %.3f, Z: %.3f) using camera default speed for profile %s...",
+			panPos, tiltPos, zoomPos, profileToken)
+	}
+
+	res, err := s.dev.CallMethod(req, s.logger)
+	if err != nil {
+		// This is an HTTP or connection level error
+		return nil, fmt.Errorf("failed to call AbsoluteMove: %w", err)
+	}
+
+	bodyBytes, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read AbsoluteMove response body: %w", err)
+	}
+
+	// this could contain a response body error from the camera
+	response := string(bodyBytes)
+	s.logger.Debugf("AbsoluteMove raw response body: %s", response)
+	return map[string]interface{}{"response": response}, nil
+}
+
+// DoCommand maps incoming commands to the appropriate ONVIF PTZ action.
+func (s *onvifPtzClient) DoCommand(_ context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
+	command, err := getString(cmd, "command")
+	if err != nil {
+		return nil, errors.New("invalid command request: 'command' key missing or not a string")
+	}
+
+	s.logger.Debugf("Received command: %s with args: %v", command, cmd)
+
+	switch strings.ToLower(command) {
+	case "get-profiles":
+		return s.handleGetProfiles()
```

Comment:
Would also be good to have a `GetCapabilities`  handler to see if and what PTZ Apis are supported by the camera?

Reply:
I like that. Thoughts on adding a new method for the POC @nicksanford ?